src/main/java/org/dependencytrack/tasks/OsvDownloadTask.java

```diff
@@ -105,32 +108,41 @@ public OsvDownloadTask() {
 
     @Override
     public void inform(Event e) {
+        if (!(e instanceof OsvMirrorEvent)) {
+            return;
+        }
 
-        if (e instanceof OsvMirrorEvent) {
-
-            if (this.ecosystems != null && !this.ecosystems.isEmpty()) {
-                for (String ecosystem : this.ecosystems) {
-                    LOGGER.info("Updating datasource with Google OSV advisories for ecosystem " + ecosystem);
-                    String url = this.osvBaseUrl + URLEncoder.encode(ecosystem, StandardCharsets.UTF_8).replace("+", "%20")
-                            + "/all.zip";
-                    HttpUriRequest request = new HttpGet(url);
-                    try (var ignoredMdcOsvEcosystem = MDC.putCloseable("osvEcosystem", ecosystem);
-                         final CloseableHttpResponse response = HttpClientPool.getClient().execute(request)) {
-                        final StatusLine status = response.getStatusLine();
-                        if (status.getStatusCode() == HttpStatus.SC_OK) {
-                            try (InputStream in = response.getEntity().getContent();
-                                 ZipInputStream zipInput = new ZipInputStream(in)) {
-                                unzipFolder(zipInput);
-                            }
-                        } else {
-                            LOGGER.error("Download failed : " + status.getStatusCode() + ": " + status.getReasonPhrase());
-                        }
-                    } catch (Exception ex) {
-                        LOGGER.error("Exception while executing Http client request", ex);
+        if (this.ecosystems == null || this.ecosystems.isEmpty()) {
+            LOGGER.info("Google OSV mirroring is disabled. No ecosystem selected.");
+            return;
+        }
+
+        for (final var ecosystem : this.ecosystems) {
+            LOGGER.info("Updating datasource with Google OSV advisories for ecosystem " + ecosystem);
+            final var url = this.osvBaseUrl + URLEncoder.encode(ecosystem, StandardCharsets.UTF_8).replace("+", "%20") + "/all.zip";
+            final var request = new HttpGet(url);
+            try (final var ignoredMdcOsvEcosystem = MDC.putCloseable("osvEcosystem", ecosystem); final var response = HttpClientPool.getClient().execute(request)) {
+                final var status = response.getStatusLine();
+                if (status.getStatusCode() != HttpStatus.SC_OK) {
+                    LOGGER.error("Download failed : " + status.getStatusCode() + ": " + status.getReasonPhrase());
+                    continue;
+                }
+
+                final var tempFile = File.createTempFile("google-osv-download", ".zip");
+                try {
+                    try (final var out = new FileOutputStream(tempFile); final var in = response.getEntity().getContent()) {
```

Comment:
Consider using `Files.newOutputStream` here, and also wrapping the resulting stream in a [`BufferedOutputStream`](https://docs.oracle.com/javase/8/docs/api/java/io/BufferedOutputStream.html) to reduce disk I/O.